### DISCUSSION TITLE
importing

### DISCUSSION THREAD

**Discussion by Stunt0265** [shlinkio/discussions#96] [Help wanted] [answered]
Question about importing:

https://shlink.io/documentation/advanced/import-short-urls/

If I want to setup a new instance but using the same custom domain, will this process still work?  In other words, the importer asks for the domain and api key.  So I want to import urls from "something.link" to a new instance that also uses "something.link".

Or should I have an instance under a different domain, like "something.info" and the import will just add another domain under that instance with the urls from "something.link"?

Thanks

**Comment by Stunt0265** [comment]
I was able to do this successfully and understand better what it's doing.  It converts the domain of the urls to whatever domain the new instance is on.  I'll need to see if there's a way to change the domain without installing a new instance.

**Comment by Stunt0265** [comment]
Is it correct to assume that if I point another domain to this same instance, that the URLs will assume this domain as well?  Interpreting this:  https://shlink.io/documentation/advanced/multiple-domains/

**Reply by acelaya** [reply]
Yes

**Comment by Stunt0265** [comment]
And the "manage domains" will just show the new domain?

**Comment by acelaya** [comment]
> If I want to setup a new instance but using the same custom domain, will this process still work?

This is the most common use case when importing from another Shlink instance.

Let's say you want to change the database engine, migrate to another server, etc., but once migrated, you want it to be exposed transparently and all existing URLs to continue working on the same domain.

The way to do this is to setup your new Shlink instance, configure it with the same domain, but not exposed via DNS yet.

Once installed, you run the import from there, using your existing domain, which still pointers to your old Shlink instance.

Once import is finished, then you update your DNS (or change your server configuration or whatever) so that you switch to the new instance.

And that's it.

**Reply by Stunt0265** [reply]
In my case, I created a new instance and new domain, and imported from the old domain.  So all the URLs use the base new domain.  I then pointed the old domain to this new instance and can created NEW short URLs with the old domain.  But is there any way to have it also allow all the imported ones using the new domain to work for the old domain as well?

**Reply by Stunt0265** [reply]
maybe doing an export from the new instance and then import back but changing the short URLs manually to the old base domain?

**Reply by acelaya** [reply]
But what's the point of the new domain? Aren't you trying to keep the same domain? Or am I missing something?

**Reply by Stunt0265** [reply]
We setup a new domain so I could work on upgrading the old one.  So there are links on the new domain my staff have set up.  Since I have this new instance up and running exactly how and where I want it, I'm trying to save myself the task or starting over.  I exported all the URLs from the new setup and can do a search and replace to change the base domain to the old base domain, and then run your import CSV tool, but it looks like it doesn't import from your export.  I'll need to tweak the columns.  Unless there's another easier way.

**Reply by Stunt0265** [reply]
And had I realized I could do the import as you described, I would have done it that way, but this is already setup now.

**Reply by Stunt0265** [reply]
What would happen if I imported the long URL and short code from the CSV file?  Would it then create short URLs for both domains?

**Comment by Stunt0265** [comment, marked as answer]
My CSV import worked so I'm good for now.  The only thing I'd like to change is the default domain for new URLs.  I don't see a way to do this in the "manage domains" section, although I do see one set as the default.

** disregard - I see how to change this.
